pyvista/plotting/mapper.py

```diff
@@ -303,19 +303,14 @@ def set_custom_opacity(
         ):
             """Set custom opacity."""
             # create a custom RGBA array to supply our opacity to
-            if opacity.size == mesh.n_points and opacity.size == mesh.n_cells:
-                if preference == 'points':
-                    rgba = np.empty((mesh.n_points, 4), np.uint8)
-                else:
-                    rgba = np.empty((mesh.n_cells, 4), np.uint8)
-            elif opacity.size == mesh.n_points:
```

Comment:
Isn't removing this logic dangerous? What about the rare edge case where the number of points matches the number of cells in the dataset?

Reply:
In the first branch where the number of points and cells matches, we have
```py
opacity.size == mesh.n_points == mesh.n_cells
```
So when we used to do 
```py
                if preference == 'points':
                    rgba = np.empty((mesh.n_points, 4), np.uint8)
                else:
                    rgba = np.empty((mesh.n_cells, 4), np.uint8)
```
on both branches we effectively ended up doing
```py
rgba = np.empty((opacity.size, 4), np.uint8)
```
because all three numbers are the same.

So when _both_ branches are true (i.e. `opacity.size == mesh.n_points` and `opacity.size == mesh.n_cells`), we're doing the same thing anyway so it doesn't matter which branch we take.

If you think this can be misleading I can add a comment to the first branch along the lines of `# includes mesh.n_points == mesh.n_cells case`. The preference only matters when we pass it on to another helper at the end.

Reply:
> If you think this can be misleading...

I think we're good here. We've just removed an unnecessary branch.